persistence/sql/src/main/java/org/infinispan/persistence/sql/AbstractSchemaJdbcStore.java

```diff
@@ -110,6 +110,9 @@ int typeWeUse(int sqlType, String typeName) {
       } else if (typeName.toUpperCase().startsWith("BOOL")) {
          // Some databases store as int32 or something similar but have the typename as BOOLEAN or some derivation
          return Types.BOOLEAN;
+      } else if (sqlType == Types.NUMERIC && scale == 0) {
+         // If scale is 0 we don't want to use float or double types
+         return Types.INTEGER;
```

Comment:
Just to be on the safe side, should we use a  BIGINT for this case or it is overkill?